src/v/transform/logging/record_batcher.cc

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "transform/logging/record_batcher.h"
+
+#include "absl/algorithm/container.h"
+#include "model/record.h"
+#include "storage/record_batch_builder.h"
+#include "transform/logging/logger.h"
+#include "utils/human.h"
+#include "utils/vint.h"
+
+namespace transform::logging {
+
+namespace detail {
+class batcher_impl {
+public:
+    batcher_impl() = delete;
+    explicit batcher_impl(size_t batch_max_bytes)
+      : _batch_max_bytes(batch_max_bytes) {}
+    ~batcher_impl() = default;
+    batcher_impl(const batcher_impl&) = delete;
+    batcher_impl& operator=(const batcher_impl&) = delete;
+    batcher_impl(batcher_impl&&) = delete;
+    batcher_impl& operator=(batcher_impl&&) = delete;
+
+    model::record_batch make_batch_of_one(iobuf k, iobuf v) {
+        return std::move(bb_init().add_raw_kv(std::move(k), std::move(v)))
+          .build();
+    }
+
+    void append(iobuf k, iobuf v) {
+        auto tmp = make_batch_of_one(std::move(k), std::move(v));
```

Comment:
Thanks for updating this, while it is not as efficient, I don't think this is the hot path and it's much more maintainable. If we see the performance being an issue we can certainly tackle improving it.

Reply:
works for me 👍